yb-voyager/cmd/importData.go

```diff
@@ -1010,6 +1010,33 @@ func executeSqlFile(file string, objType string, skipFn func(string, string) boo
 	}
 }
 
+func getIndexName(sqlQuery string, indexName string) (string, error) {
+	sqlQuery = strings.TrimSpace(sqlQuery)
+	if !strings.HasPrefix(strings.ToUpper(sqlQuery), "CREATE INDEX") {
+		err := fmt.Errorf("not a CREATE INDEX statement")
+		return "", err
+	}
+
+	// Return the index name itself if it is aleady qualified with schema name
+	if len(strings.Split(indexName, ".")) == 2 {
+		return indexName, nil
+	}
+
+	fullyQualifiedIndexName := indexName
+
+	splits := strings.Fields(sqlQuery)
+
+	for index, split := range splits {
+		if strings.EqualFold(split, "ON") {
+			tableName := splits[index+1]
+			schemaName := getTargetSchemaName(tableName)
+			fullyQualifiedIndexName = schemaName + "." + indexName
+			break
+		}
+	}
+	return fullyQualifiedIndexName, nil
```

Comment:
return "", fmt.Errorf("could not find `ON` keyword in the CREATE INDEX statement")

Reply:
Oh yeah my bad. Realised it later that `getTargetSchema` returns public in case of no schema name.

Reply:
Oh yeah my bad. I was applying a condition inside the for loop before to handle the case when schema name is not attached but then realised that getTargetSchema returns public in that case as well. Forgot to change this line.